### PR TITLE
java to Kotlin MapboxNavigationOptions

### LINKED ISSUE #2094: MapboxNavigationOptions conversion to Kotlin
The MapboxNavigationOptions class is being converted to a data class

### PR DESCRIPTION
**Pull request by olegzil**
## Description
Closes #2094 
Converts MapboxNavigationOptions to a data class

### Goal
Convert to Kotlin
## Testing

Please describe the manual tests that you ran to verify your changes

- [x ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @JvmField val isFromNavigationUi: Boolean = false,
+    @JvmField val isDebugLoggingEnabled: Boolean = false,
+    @JvmField val navigationNotification: NavigationNotification? = null,
+    @JvmField val roundingIncrement: Int = ROUNDING_INCREMENT_FIFTY,
+    @JvmField val timeFormatType: Int = NavigationTimeFormat.NONE_SPECIFIED,
+    @JvmField val navigationLocationEngineIntervalLagInMilliseconds: Int = NAVIGATION_LOCATION_ENGINE_INTERVAL_LAG,
+    @JvmField val defaultNotificationColorId: Int = R.color.mapboxNotificationBlue
+) {
+    companion object {
+        fun build() = MapboxNavigationOptions()
```
Comment (abhishek1508) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt]: Can we add `@JvmStatic`. That way we can refrain from using `Companion` everywhere. 
Makes the code more readable

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationNotificationProvider.java [hunk 2]
```diff
@@ -34,8 +34,8 @@ void shutdown(Context applicationContext) {
 
   private NavigationNotification buildNotificationFrom(Context applicationContext, MapboxNavigation mapboxNavigation) {
     MapboxNavigationOptions options = mapboxNavigation.options();
-    if (options.navigationNotification() != null) {
-      return options.navigationNotification();
+    if (options.navigationNotification != null) {
```
Comment (abhishek1508) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationNotificationProvider.java]: Can we use the let and elvis operator here

Comment (olegzil) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationNotificationProvider.java]: 👍

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt (hunk 1 above)
Comment (olegzil) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt]: 👍

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    @JvmField val isDebugLoggingEnabled: Boolean = false,
+    @JvmField val navigationNotification: NavigationNotification? = null,
+    @JvmField val roundingIncrement: Int = NavigationConstants.ROUNDING_INCREMENT_FIFTY,
+    @JvmField val timeFormatType: Int = NavigationTimeFormat.NONE_SPECIFIED,
+    @JvmField val navigationLocationEngineIntervalLagInMilliseconds: Int = NavigationConstants.NAVIGATION_LOCATION_ENGINE_INTERVAL_LAG,
+    @JvmField val defaultNotificationColorId: Int = R.color.mapboxNotificationBlue
+
+) {
+    fun toBuilder() = Builder()
+    fun get_navigationNotification() = navigationNotification
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt]: Why are we using snake case nomenclature? 🤔

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteProcessorRunnableTest.java [hunk 4]
```diff
@@ -187,7 +187,7 @@ public void onRun_legIndexDoesNotIncrementOnLegCompletionWithAutoIncrementDisabl
   private RouteProcessorRunnable buildRouteProcessorRunnableWith(MapboxNavigator navigator,
                                                                  NavigationRouteProcessor processor,
                                                                  NavigationStatus status, DirectionsRoute route) {
-    MapboxNavigationOptions options = MapboxNavigationOptions.builder().build();
+    MapboxNavigationOptions options =  (new MapboxNavigationOptions.Builder()).build();
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteProcessorRunnableTest.java]: NIT extra space

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationNotificationProviderTest.java [hunk 5]
```diff
@@ -62,7 +62,7 @@ public void onShutdown_onNavigationStoppedIsCalled() {
   private MapboxNavigation buildNavigationWithNotificationOptions(NavigationNotification notification) {
     MapboxNavigation mapboxNavigation = mock(MapboxNavigation.class);
     MapboxNavigationOptions options = mock(MapboxNavigationOptions.class);
-    when(options.navigationNotification()).thenReturn(notification);
+    when(options.get_navigationNotification()).thenReturn(notification);
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationNotificationProviderTest.java]: `NavigationNotificationProviderTest` stubs are broken making tests to fail 👀 

```
com.mapbox.services.android.navigation.v5.internal.navigation.NavigationNotificationProviderTest > updateNavigationNotification FAILED
    java.lang.IllegalStateException at NavigationNotificationProviderTest.java:27

com.mapbox.services.android.navigation.v5.internal.navigation.NavigationNotificationProviderTest > onShutdown_onNavigationStoppedIsCalled FAILED
    java.lang.IllegalStateException at NavigationNotificationProviderTest.java:54

com.mapbox.services.android.navigation.v5.internal.navigation.NavigationNotificationProviderTest > updateNavigationNotification_doesNotUpdateAfterShutdown FAILED
    java.lang.IllegalStateException at NavigationNotificationProviderTest.java:40
```

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteRefresherTest.java [hunk 6]
```diff
@@ -26,8 +26,8 @@ public void checksShouldRefreshIfIntervalHasPassed() {
     long aRefreshIntervalInMilliseconds = 1;
     MapboxNavigationOptions mockedMapboxNavigationOptions = mock(MapboxNavigationOptions.class);
     when(mockedMapboxNavigation.options()).thenReturn(mockedMapboxNavigationOptions);
-    when(mockedMapboxNavigationOptions.refreshIntervalInMilliseconds()).thenReturn(aRefreshIntervalInMilliseconds);
-    when(mockedMapboxNavigationOptions.enableRefreshRoute()).thenReturn(true);
+    when(mockedMapboxNavigationOptions.get_refreshIntervalInMilliseconds()).thenReturn(aRefreshIntervalInMilliseconds);
+    when(mockedMapboxNavigationOptions.get_enableRefreshRoute()).thenReturn(true);
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteRefresherTest.java]: Same as above re: stubs are broken making this test to fail 👀 

```
com.mapbox.services.android.navigation.v5.internal.navigation.RouteRefresherTest > checksShouldRefreshIfIntervalHasPassed FAILED
    java.lang.AssertionError at RouteRefresherTest.java:37
```

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/MapboxNavigationActivity.java [hunk 7]
```diff
@@ -96,7 +96,7 @@ public void onNavigationReady(boolean isRunning) {
     options.navigationListener(this);
     extractRoute(options);
     extractConfiguration(options);
-    options.navigationOptions(MapboxNavigationOptions.builder().build());
+    options.navigationOptions( (new MapboxNavigationOptions.Builder()).build());
```
[the remark this answers is not in the document]
Comment (olegzil) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/MapboxNavigationActivity.java]: Agreed @Guardiola31337, data classes are the way to go. The MapboxNavigationsOptions is going to be one after the next push

**Comment by Guardiola31337** [comment]
Could we rebase from `master` and :eyes: what's failing at the moment?

**Comment by Guardiola31337** [comment]
Tests are failing ❌ 👀 

```
com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationTest > bannerInstructionMilestone_onInitializationDoesGetAdded FAILED
    junit.framework.AssertionFailedError at MapboxNavigationTest.java:70

com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationTest > voiceInstructionMilestone_onInitializationDoesGetAdded FAILED
    junit.framework.AssertionFailedError at MapboxNavigationTest.java:61
```

**Pull request merged**